Python/pytime.c

```diff
@@ -401,7 +401,11 @@ static int
 _PyTime_FromObject(_PyTime_t *t, PyObject *obj, _PyTime_round_t round,
                    long unit_to_ns)
 {
-    if (PyFloat_Check(obj)) {
+    if (!PyFloat_Check(obj) && !PyLong_Check(obj)) {
```

Comment:
I don't think this is correct. In the else case below it calls PyLong_AsLongLong, which can handle value which are not PyLongs as long as they can be converted to a PyLong via _PyNumber_Index. 

If this change didn't break any unit tests, then a test is probably missing.

Reply:
ah, I see you've commented on that. Sorry for the noise.